src/Autofac/Core/Lifetime/LifetimeScope.cs

```diff
@@ -75,23 +74,14 @@ public class LifetimeScope : Disposable, ISharingLifetimeScope, IServiceProvider
         /// <param name="componentRegistry">Components used in the scope.</param>
         /// <param name="parent">Parent scope.</param>
         protected LifetimeScope(IComponentRegistry componentRegistry, LifetimeScope parent, object tag)
-            : this(componentRegistry, parent, null, tag)
         {
-        }
-
-        /// <summary>
-        /// Initializes a new instance of the <see cref="LifetimeScope"/> class.
-        /// </summary>
-        /// <param name="tag">The tag applied to the <see cref="ILifetimeScope"/>.</param>
-        /// <param name="componentRegistry">Components used in the scope.</param>
-        /// <param name="parent">Parent scope.</param>
-        /// <param name="tracer">A tracer instance for this scope.</param>
-        protected LifetimeScope(IComponentRegistry componentRegistry, LifetimeScope parent, IResolvePipelineTracer? tracer, object tag)
-            : this(componentRegistry, tag)
-        {
-            _tracer = tracer;
+            ComponentRegistry = componentRegistry ?? throw new ArgumentNullException(nameof(componentRegistry));
+            Tag = tag ?? throw new ArgumentNullException(nameof(tag));
             _parentScope = parent ?? throw new ArgumentNullException(nameof(parent));
+
+            _sharedInstances[SelfRegistrationId] = this;
             RootLifetimeScope = _parentScope.RootLifetimeScope;
+            this.DiagnosticSource = _parentScope.DiagnosticSource;
```

Comment:
Remaining `this.`

Reply:
Removed all `this.` from the src/** set of files.